splunk-otel-android/src/main/java/com/splunk/rum/AppStartupTimer.java

```diff
@@ -44,15 +57,76 @@ Span start(Tracer tracer) {
         return appStart;
     }
 
+    /** Called when Activity is created. */
+    void startUiInit() {
+        if (uiInitStarted || isStartedFromBackground) {
+            return;
+        }
+        uiInitStarted = true;
+        if (firstPossibleTimestamp + MAX_TIME_TO_UI_INIT < startupClock.now()) {
+            Log.d(SplunkRum.LOG_TAG, "Max time to UI init exceeded");
+            uiInitTooLate = true;
+            clear();
+        }
+    }
+
+    void setCompletionCallback(Runnable completionCallback) {
+        this.completionCallback = completionCallback;
+    }
+
     void end() {
-        if (overallAppStartSpan != null) {
+        if (overallAppStartSpan != null && !uiInitTooLate && !isStartedFromBackground) {
```

Comment:
Are we guaranteed that `end()` is called from the same thread that might have assigned those fields? If not, should we make all those volatile as well?

Reply:
This should also be called from the UI thread. I make `isStartedFromBackground` volatile, could make others also if you feel it is better that way.

Reply:
`isStartedFromBackground` is only accessed from the UI thread, and does not need to be volatile, right?
I'd say that we either make all of these fields volatile, or make them all non-volatile and leave a comment saying sth like "these 3 booleans are only accessed from the UI thread"

Reply:
Makes sense, reverted the previous change and added comment instead.